src/coreclr/tools/Common/JitInterface/LoongArch64PassStructInRegister.cs

```diff
@@ -55,16 +55,17 @@ public static uint GetLoongArch64PassStructInRegisterFlags(TypeDesc typeDesc)
             int fieldIndex = 0;
             foreach (FieldDesc field in typeDesc.GetFields())
             {
-                if (fieldIndex > 1)
+                if (field.IsStatic)
                 {
-                    return (uint)StructFloatFieldInfoFlags.STRUCT_NO_FLOAT_FIELD;
+                    continue;
                 }
-                else if (field.IsStatic)
+
+                if ((numIntroducedFields == 2) && (field.FieldType.Category == TypeFlags.Class))
                 {
-                    continue;
+                    return (uint)StructFloatFieldInfoFlags.STRUCT_NO_FLOAT_FIELD;
```

Comment:
Should this case be rather handled in the `default:` case of the switch statement below?

Reply:
I will amend this, thank you!